docs/FormatsAndConventions.adoc

```diff
@@ -0,0 +1,307 @@
+`JavaCodingStyleBasics.html` and `HtmlCodingStyleBasics.html`
+* For documents, try to make the file name match the document title as much as possible.
+
+== Issue
+
+* Issue title should be concise yet descriptive.
+For example, instead of `Newbie question, please help`, use `How do I set up git to ignore test files?`
+* The phrasing should match the main purpose of the issue.
+For example, if it is a bug report, the issue title should sound like a bug report
+(e.g `Option 'other' is missing from the dropdown`) instead of a feature request
+(e.g. `Add 'other' option to the dropdown`).
+
+== Merge commit
+
+This format is only for commits merging a PR branch to `master` branch.
+
+Format: *`[#IssueNumber] Issue Title (#PrNumber)`* +
+e.g. `[#5958] Error alert email has very long subject (#6580)`
+
+[NOTE,caption=Rationale]
+====
+This format allows easy traceability among a merge commit, the issue it fixes, and the PR that fixed it.
+Having the issue name tells us what the commit is about without having to look it up in GitHub issue tracker.
+====
+
+== PR
+
+Format: *`IssueTitle #IssueNumber`* +
+e.g. `Error alert email has very long subject #5958`
+
+[NOTE,caption=Rationale]
+====
+Duplicating issue title in PR title is for easy tracing between PRs and issues,
+to compensate for GitHub's lack of strong linking between the two.
+Assume there is an invisible prefix in front of the PR title `Fixes issue : ...`
+====
+
+== References to code elements
+
+Follow these conventions when referring to code elements *from a non-code context* e.g. when referring to a function
+name from a commit message.
+
+[NOTE]
+====
+The objective is to be as concise as possible without being ambiguous. Therefore, omit optional details when
+those details are not pertinent to the context.
+
+Refer to the <<CodingStandards.adoc#,respective coding standards>> for conventions on how to refer to code elements
+*from code contexts* e.g. when referring to a function from a code comment.
+====
+
+=== Java
+
+* Variables: `package.class#variable` (optional: `package`)
++
+====
+Examples:
+
+* `seedu.address.data.Person#name`
+* `Person#name` -- optional parts omitted
+====
+
+* Methods: `package.class#method(paramTypes):returnType` (optional: `package`, `returnType`)
++
+====
+Examples:
+
+* `seedu.address.data.Person#getName(boolean):String`
+* `Person#getName(boolean)` -- optional parts omitted
+====
+
+* If including `paramTypes` pushes against a severe length constraint (e.g. in the commit message title),
+it can be replaced with `...` as long as it is not ambiguous.
++
+====
+For example,
+`Person#add(...)` is acceptable in a commit message title (which is limited to 50 chars) in place of
+`Person#add(String, boolean)`.
+====
+
+* The `class` part can be omitted if it is clear from the context.
++
+====
+For example, the commit message title
+`AbstractPerson: remove add(int) method` is acceptable in place of
+`AbstractPerson: remove AbstractPerson#add(int) method`.
+====
```

Comment:
Perhaps we can use this 'box style' for all examples everywhere. Probably out of scope for this PR.